### DISCUSSION TITLE
sort result by most read

### DISCUSSION THREAD

**Discussion by mc-stephen** [directus/directus#15344] [Archive]
Hello, so if I want all the blog posts I have in my Directus using an API and then sorting it by most current post, I can do it like this 
`https://example.com/items/Axxellanceblog?sort[]=-date_created`

but what I don't know how to do is sort my article by most read, for example, the most read article with being first, and the less read with be at the bottom.

I will really appreciate anybody's help on this thanks in advance

**Comment by rijkvanzanten** [comment]
Heya! Directus by default doesn't keep track of how often an item was "read", so there's nothing to sort by. That being said, you should be able to add a numeric field "read_count" (name can be whatever you want), populate that through the API when somebody opens the article, and then sort by that 🙂

**Reply by mc-stephen** [reply]
hmmm, if am not misunderstanding what you are saying, then you mean I should add a field called "read_count" (or whatever I want), and then when a request is being sent to retrieve that article then I should increment it by 1 every time. so that I can sort it by the count...

if that is so then how will I sort it out, will  `https://example.com/items/article?sort=read_count` do it for me?

thanks for the first answer.

**Reply by mc-stephen** [reply]
wow, nice I have done it, thanks for the idea, I really appreciate it a lot. :)
